// eslint-plugin-10x/dist/test/auto-import.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const ava_1 = __importDefault(require("ava"));
const eslint_ava_rule_tester_1 = __importDefault(require("eslint-ava-rule-tester"));
const auto_import_1 = __importDefault(require("../rules/auto-import"));
const ruleTester = eslint_ava_rule_tester_1.default(ava_1.default, {
    env: {
        es6: true
    },
    parserOptions: {
        sourceType: 'module',
        ecmaFeatures: {
            jsx: true
        }
    }
});
ruleTester.run('tenx/auto-import', auto_import_1.default, {
    invalid: [
        {
            code: 'const Container = styled.div``',
            errors: [
                {
                    ruleId: 'tenx/auto-import',
                    message: '\'styled\' is not defined. Run --fix to add `import styled from "styled-components";`'
                }
            ],
            options: [
                {
                    imports: {
                        styled: 'import styled from "styled-components";'
                    }
                }
            ],
            output: 'import styled from "styled-components";\nconst Container = styled.div``'
        },
        {
            code: 'const a = "hi"',
            errors: [
                {
                    ruleId: 'tenx/auto-import',
                    message: 'Rule tenx/auto-import is enabled but there are no replacements specified. Get started by specifying an object as an option to the ESLint rule: ["error", {imports: {sortBy: `import sortBy from "lodash/sortBy"`}}]'
                }
            ],
            options: [
                {
                    imports: {}
                }
            ],
            output: 'const a = "hi"'
        },
        {
            code: 'const Container = styled.div``',
            errors: [
                {
                    ruleId: 'tenx/auto-import',
                    message: '\'styled\' is not defined. Run --fix to add `import styled from "polished";`'
                }
            ],
            options: [
                {
                    imports: {
                        styled: 'import styled from "polished";'
                    }
                }
            ],
            output: 'import styled from "polished";\nconst Container = styled.div``'
        },
        {
            code: 'export default () => <View></View>',
            errors: [
                {
                    ruleId: 'tenx/auto-import',
                    message: '\'View\' is not defined. Run --fix to add `import {View} from "react-native";`'
                }
            ],
            options: [
                {
                    imports: {
                        View: 'import {View} from "react-native";'
                    }
                }
            ],
            output: 'import {View} from "react-native";\nexport default () => <View></View>'
        },
        {
            code: 'const a = b',
            errors: [
                {
                    ruleId: 'tenx/auto-import',
                    message: "'b' is not defined."
                }
            ],
            options: [
                {
                    imports: {
                        styled: 'import styled from "styled-components";'
                    }
                }
            ]
        }
    ],
    valid: [
        {
            code: `
        import styled from 'styled-components';
        const Container = styled.div;
    `,
            options: [
                {
                    imports: {
                        styled: 'import styled from "styled-components";'
                    }
                }
            ]
        }
    ]
});
