// eslint-plugin-10x/dist/test/react-in-scope.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const ava_1 = __importDefault(require("ava"));
const eslint_ava_rule_tester_1 = __importDefault(require("eslint-ava-rule-tester"));
const react_in_scope_1 = __importDefault(require("../rules/react-in-scope"));
const ruleTester = eslint_ava_rule_tester_1.default(ava_1.default, {
    env: {
        es6: true
    },
    parserOptions: {
        sourceType: 'module',
        ecmaFeatures: {
            jsx: true
        }
    }
});
ruleTester.run('tenx/react-in-scope', react_in_scope_1.default, {
    invalid: [
        {
            code: 'export default () => <div></div>',
            errors: [
                {
                    ruleId: 'tenx/react-in-scope',
                    message: "'React' must be in scope when using JSX"
                }
            ],
            output: 'import React from "react";\nexport default () => <div></div>'
        }
    ],
    valid: ['import React from "react";\nexport default () => <div></div>']
});
